recommendation/model/post_rec.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.metrics.pairwise import cosine_similarity
+from scipy.sparse import csr_matrix, hstack
+
+def get_matrix(posts, n):
+    n = n+1    
+    all_posts = [None]*n
+
+    for post in posts:
+        all_posts[post['id']] = post
+
+    all_categories = [""]*n
+
+    all_descriptions = [""]*n
+    # insert date due
+    all_titles = [""]*n
+    all_image_prompts = [""]*n
+    all_user_rating = [0]*n
+    all_bid_counts = [0]*n
+    all_likes = [0]*n
+
+
+
+    for idx in range(0, len(all_posts)):
```

Comment:
curious, could we have this data be preprocessed / stored in the db for each post? could save computation so that we don't have to calculate it each time! but it is a space vs. compute issue, so dependent on what you want to prioritize!